osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SpinnerDisc.cs

```diff
@@ -73,6 +73,8 @@ public bool Complete
             }
         }
 
+        public bool AutoSpin { get; set; } = false;
```

Comment:
`HitObject` implementations (and their drawables) should not know about mods. You will need to move this logic to inside your mod class.